src/traces/heatmap/plot.js

```diff
@@ -337,7 +337,14 @@ function plotOne(gd, plotinfo, cd) {
 
     if(zsmooth) { // best or fast, works fastest with imageData
         var pxIndex = 0,
+            pixels;
+
+        try {
             pixels = new Uint8Array(imageWidth * imageHeight * 4);
```

Comment:
Thanks for checking!

I'm curious. Would a fallback be faster than `try-catch`:

```js
var pixels = (new Uint8Array || new Array)(imageWidth * imageHeight * 4);
```